consul/resource_consul_service_splitter_config_entry.go

```diff
@@ -0,0 +1,255 @@
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				parts := strings.Split(d.Id(), "/")
+				var kind, name, partition, namespace string
+				switch len(parts) {
+				case 2:
+					kind = parts[0]
+					name = parts[1]
+				case 4:
+					partition = parts[0]
+					namespace = parts[1]
+					kind = parts[2]
+					name = parts[3]
+				default:
+					return nil, fmt.Errorf(`expected path of the form "<kind>/<name>" or "<partition>/<namespace>/<kind>/<name>"`)
+				}
+
+				d.SetId(fmt.Sprintf("%s-%s", kind, name))
+				sw := newStateWriter(d)
+				sw.set("kind", kind)
+				sw.set("name", name)
+				sw.set("partition", partition)
+				sw.set("namespace", namespace)
+
+				err := sw.error()
+				if err != nil {
+					return nil, err
+				}
+
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+		Schema: serviceSplitterConfigEntrySchema,
+	}
+}
+
+func resourceConsulServiceSplitterConfigEntryUpdate(d *schema.ResourceData, meta interface{}) error {
+	client, qOpts, wOpts := getClient(d, meta)
+	configEntries := client.ConfigEntries()
+
+	name := d.Get("name").(string)
+
+	configMap := make(map[string]interface{})
+	configMap["kind"] = KindServiceSplitter
+
+	configMap["name"] = name
+
+	kind := configMap["kind"].(string)
+	err := d.Set("kind", kind)
+
+	if err != nil {
+		return err
+	}
+
+	var attributes []string
+
+	for key, _ := range serviceSplitterConfigEntrySchema {
+		attributes = append(attributes, key)
+	}
+
+	for _, attribute := range attributes {
+		configMap[attribute] = d.Get(attribute)
+	}
+
+	formattedMap, err := formatKeys(configMap, formatKey)
+	if err != nil {
+		return err
+	}
+
+	configEntry, err := makeServiceSplitterConfigEntry(name, formattedMap.(map[string]interface{}), wOpts.Namespace, wOpts.Partition)
+	if err != nil {
+		return err
+	}
+
+	if _, _, err := configEntries.Set(configEntry, wOpts); err != nil {
+		return fmt.Errorf("failed to set '%s' config entry: %v", name, err)
+	}
+	_, _, err = configEntries.Get(configEntry.GetKind(), configEntry.GetName(), qOpts)
+	if err != nil {
+		if strings.Contains(err.Error(), "Unexpected response code: 404") {
+			return fmt.Errorf(`failed to read config entry after setting it.
+This may happen when some attributes have an unexpected value.
+Read the documentation at https://www.consul.io/docs/agent/config-entries/%s.html
+to see what values are expected`, configEntry.GetKind())
+		}
+		return fmt.Errorf("failed to read config entry: %v", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s-%s", kind, name))
+	return resourceConsulServiceSplitterConfigEntryRead(d, meta)
+}
+
+func resourceConsulServiceSplitterConfigEntryRead(d *schema.ResourceData, meta interface{}) error {
+	client, qOpts, _ := getClient(d, meta)
+	configEntries := client.ConfigEntries()
+	configKind := d.Get("kind").(string)
+	configName := d.Get("name").(string)
+
+	fixQOptsForConfigEntry(configName, configKind, qOpts)
+
+	_, _, err := configEntries.Get(configKind, configName, qOpts)
```

Comment:
You need to use the result from `configEntries.Get()` to update the Terraform state with d.Set() or stateWriter.set()